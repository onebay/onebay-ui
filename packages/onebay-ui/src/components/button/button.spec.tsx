import { mount } from '@vue/test-utils'
import Button from './index'

const factory = (values = {}, slots = { default: ['click me'] }) => {
  return mount(Button, {
    slots,
    props: { ...values }
  })
}

describe('test Button', () => {
  it('render Button -- props size(normal)', () => {
    const wrapper = factory({
      size: 'normal'
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Button -- props size(small)', () => {
    const wrapper = factory({
      size: 'small'
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Button -- props size(primary)', () => {
    const wrapper = factory({
      size: 'primary'
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Button -- props size(secondary)', () => {
    const wrapper = factory({
      size: 'secondary'
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Button -- props circle', () => {
    const wrapper = factory({
      circle: true
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Button -- props full', () => {
    const wrapper = factory({
      full: true
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Button -- props loading', () => {
    const wrapper = factory({
      loading: true
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Button -- props disabled', () => {
    const wrapper = factory({
      disabled: true
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Button Event', () => {
  it('Button trigger click event', () => {
    const wrapper = factory()
    wrapper.trigger('click')
    expect(wrapper.emitted()).toHaveProperty('click')
    expect(wrapper.emitted('click')).toHaveLength(1)
  })
  it('Button disabled, not to trigger click event', () => {
    const wrapper = factory({
      disabled: true
    })
    expect(!wrapper.emitted().hasOwnProperty('click')).toBe(true)
  })
})
