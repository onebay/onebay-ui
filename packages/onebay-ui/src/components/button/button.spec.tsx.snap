// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button render Button -- props circle 1`] = `
<div
  class="ob-button ob-button--normal ob-button--circle"
>
  <!---->
  <div
    class="ob-button__text"
  >
    
    click me
    
  </div>
</div>
`;

exports[`test Button render Button -- props disabled 1`] = `
<div
  class="ob-button ob-button--normal ob-button--disabled"
>
  <!---->
  <div
    class="ob-button__text"
  >
    
    click me
    
  </div>
</div>
`;

exports[`test Button render Button -- props full 1`] = `
<div
  class="ob-button ob-button--normal ob-button--full"
>
  <!---->
  <div
    class="ob-button__text"
  >
    
    click me
    
  </div>
</div>
`;

exports[`test Button render Button -- props loading 1`] = `
<div
  class="ob-button ob-button--icon ob-button--normal"
>
  <div
    class="ob-button__icon"
  >
    <div
      class="ob-loading"
      style=""
    >
      <div
        class="ob-loading__ring"
      />
      <div
        class="ob-loading__ring"
        style=""
      />
      <div
        class="ob-loading__ring"
        data-id="[object Object]"
      />
    </div>
  </div>
  <div
    class="ob-button__text"
  >
    
    click me
    
  </div>
</div>
`;

exports[`test Button render Button -- props size(normal) 1`] = `
<div
  class="ob-button ob-button--normal"
>
  <!---->
  <div
    class="ob-button__text"
  >
    
    click me
    
  </div>
</div>
`;

exports[`test Button render Button -- props size(primary) 1`] = `
<div
  class="ob-button"
>
  <!---->
  <div
    class="ob-button__text"
  >
    
    click me
    
  </div>
</div>
`;

exports[`test Button render Button -- props size(secondary) 1`] = `
<div
  class="ob-button"
>
  <!---->
  <div
    class="ob-button__text"
  >
    
    click me
    
  </div>
</div>
`;

exports[`test Button render Button -- props size(small) 1`] = `
<div
  class="ob-button ob-button--small"
>
  <!---->
  <div
    class="ob-button__text"
  >
    
    click me
    
  </div>
</div>
`;
